common/app-starters-task-launch-request-common/src/main/java/org/springframework/cloud/stream/app/tasklaunchrequest/TaskLauncherRequestAutoConfiguration.java

```diff
@@ -0,0 +1,161 @@
+/*
+ * Copyright 2018 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.stream.app.tasklaunchrequest;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Function;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.cloud.task.launcher.TaskLaunchRequest;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.messaging.Message;
+import org.springframework.messaging.MessageHeaders;
+import org.springframework.messaging.support.MessageBuilder;
+import org.springframework.util.Assert;
+import org.springframework.util.MimeTypeUtils;
+
+/**
+ * @author David Turanski
+ **/
+@Configuration
+@EnableConfigurationProperties(TaskLaunchRequestProperties.class)
+public class TaskLauncherRequestAutoConfiguration {
+
+	private static final Log log = LogFactory.getLog(TaskLaunchRequestTransformer.class);
+
+	private final TaskLaunchRequestProperties taskLaunchRequestProperties;
+
+	public TaskLauncherRequestAutoConfiguration(TaskLaunchRequestProperties taskLaunchRequestProperties) {
+		this.taskLaunchRequestProperties = taskLaunchRequestProperties;
+	}
+
+	@Bean
+	public TaskLaunchRequestTransformer taskLaunchRequestTransformer(
+		TaskLaunchRequestTypeProvider taskLaunchRequestTypeProvider) {
+		return message -> {
+			switch (taskLaunchRequestTypeProvider.taskLaunchRequestType()) {
+			case DATAFLOW:
+				return dataflowTaskLaunchRequest().apply(message);
```

Comment:
Do we really have to have a new `Function` instance for each request message?
You don't call `@Bean` method here, so that `standaloneTaskLaunchRequest()` is going to return a new `Function` each time we call it.

Reply:
Left over from the original Functional implementation. Good point